### PR TITLE
chore: changed the outDir to entry test

### PR DESCRIPTION
**Pull request by anikethsaha**
**What kind of change does this PR introduce?**

refactoring/bugfix

**Did you add tests for your changes?**
NA

**If relevant, did you update the documentation?**
NA

**Summary**
the current matching method, the test is failing in windows as the `Output Director` is as follows
```
\\Path-to\\webpack-cli\\test\\entry\\bin
``` 
and this is been matched with `entry/bin` which is failing .
So if we refactored the current to just check with the last two directory, we can make it to pass

**Does this PR introduce a breaking change?**

No

**Other information**
Also disabled the `@typescript-eslint/explicit-function-return-type` for `test-utils` file as its causing to failed while committing.

### CODE REVIEW

**Comment by anikethsaha** [comment]
Actually, the current test is failing in windows, as the `Output Directory` is having path using `\\`  and also it is giving the whole relative path from the drive, not just the last basename, so I did this.
Also please check I used the node default separator for splitting in the latest commit

**Comment by anikethsaha** [comment]
cc @evenstensberg @ematipico

File: test/entry/entry-single-arg.test.js [hunk 1]
```diff
@@ -10,7 +10,9 @@ describe('single entry flag', () => {
         const { stdout, stderr } = run(__dirname);
         const summary = extractSummary(stdout);
         const outputDir = 'entry/bin';
-        expect(summary['Output Directory']).toContain(outputDir);
+        const outDirectoryFromCompiler = summary['Output Directory'].split(sep);
+        const outDirToMatch = outDirectoryFromCompiler.slice(outDirectoryFromCompiler.length - 2, outDirectoryFromCompiler.length).join('/');
```
Comment (ematipico) [inline test/entry/entry-single-arg.test.js]: `.join('/')` is not correct. Windows don't have `/` in their directory paths

File: test/entry/entry-single-arg.test.js [hunk 2]
```diff
@@ -10,7 +10,9 @@ describe('single entry flag', () => {
...
+        expect(outDirToMatch).toContain(outputDir);
         expect(stderr).toContain('Entry module not found');
         stat(resolve(__dirname, './bin'), (err, stats) => {
```
Comment (ematipico) [inline test/entry/entry-single-arg.test.js]: `./bin` is not correct. It should be just `bin`

File: test/entry/entry-single-arg.test.js (hunk 1 above)
Comment (anikethsaha) [reply test/entry/entry-single-arg.test.js]: I just used `'/'` to make it in the format of the `outputDir` i.e `entry/bin` .

File: test/entry/entry-single-arg.test.js [hunk 3]
```diff
@@ -10,7 +10,12 @@ describe('single entry flag', () => {
         const { stdout, stderr } = run(__dirname);
         const summary = extractSummary(stdout);
         const outputDir = 'entry/bin';
-        expect(summary['Output Directory']).toContain(outputDir);
+        const outDirectoryFromCompiler = summary['Output Directory'];
```
[the remark this answers is not in the document]
Comment (anikethsaha) [reply test/entry/entry-single-arg.test.js]: I used the `node` method, `sep`

File: test/entry/entry-single-arg.test.js (hunk 2 above)
Comment (anikethsaha) [reply test/entry/entry-single-arg.test.js]: done

File: test/entry/entry-single-arg.test.js (hunk 1 above)
Comment (ematipico) [reply test/entry/entry-single-arg.test.js]: Yes, I know. Webpack under the hoods takes care of it. And when it's parsed, it's transformed with the path convention used by the operative system.

The use `path` utility to deal with paths.

**Pull request merged**